### DISCUSSION TITLE
trojan-gfw instead of Shadowsocks

### DISCUSSION THREAD

**Discussion by IntergalacticQuasar** [Ysurac/openmptcprouter#2486] [Q&A]
Hi there,

Just was wondering would it be possible to configure openmptcprouter to use different VPN protocols other than Shadowsocks? I live in a country that the Internet is heavily censored, and VPN servers are actively monitored and blacklisted by ISPs, so hiding the VPN server and tunnel is very beneficial. 

From my research, Shadowsocks server can be detected by active probing, and get filtered. [trojan-gfw](https://github.com/trojan-gfw/trojan) is far more stealth and cannot be easily detected by active probing. 

How difficult (or easy) would it be to make openmptcprouter work with trojan-gfw protocol?

P.S.: Not sure whether this is useful or not, but by using [tun2socks](https://github.com/xjasonlyu/tun2socks), I am able to transform a trojan-gfw to a TUN interface, but not sure how to make this work with openmptcprouter.

Thanks

**Comment by Ysurac** [comment]
There 2 proxy available on OpenMPTCProuter for now : Shadowsocks-libev (default proxy) and v2ray. There is also obfuscating plugins available for shadowsocks.

To use a new proxy like trojan-gfw, it must work with MPTCP. Not sure if it's the case or not with trojan-gfw.
